### DISCUSSION TITLE
Where is the Copilot Chat history located in local file system?

### DISCUSSION THREAD

**Discussion by TaihouKai** [community/community#69740] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I know that we can use "Chat: Export Session" to export chat history of some session (as described in https://github.com/orgs/community/discussions/57190), but where are those history files stored in local file system? I'd like to access these files by my own.

**Comment by TaihouKai** [comment, marked as answer]
Nevermind, I figured it out.
It's in `Code\User\workspaceStorage\<alphanumerical>\state.vscdb` -> `memento/interactive-session` & `interactive.sessions`

**Reply by stellarpower** [reply]
How do I get this data out, sorry? Code wouldn't reopen my dev container and then nuked it, so now I have lost quite a bit, including the message history. I have found the database, some don't have interactive-session, and others are an empty list. Thanks

**Reply by KhalfaniW** [reply]
I think each workspace has its own chat history collection. I extracted the chats for one workspace using a script like below in the workspace dir with the `Code\User\workspaceStorage\<alphanumerical>\state.vscdb` file.

This will create a group of files in the dir that is similar to the json you would get if you ran `Chat: Export Chat...` command in vscode. It just with a few extra properties. 


``` bash
#!/bin/bash
#extract the history to json
sqlite3 ./state.vscdb "SELECT value FROM ItemTable WHERE key = 'interactive.sessions';" > interactive.sessions.json
# Read the JSON file and split each object into a separate file
jq -c '.[]' interactive.sessions.json | while read -r element; do
    # Extract the customTitle and sanitize it for use as a filename
    customTitle=$(echo $element | jq -r '.customTitle' | tr -cd '[:alnum:][:space:]._-' | tr ' ' '_')
    # Ensure the filename is not empty
    if [ -z "$customTitle" ]; then
        customTitle="untitled"
    fi
    
    # Write the element to a file named after the sanitized customTitle
    echo "$element" | jq '.' > "${customTitle}.json"
done
```

`$ bash extract.sh `



_________________________
I'm not sure what `memento/interactive-session` is but you can also export it using

`sqlite3 ./state.vscdb "SELECT value FROM ItemTable WHERE key = 'memento/interactive-session';" > memento_interactive-session.json`

@stellarpower

**Comment by Stabledog** [comment]
It seems like a pretty big oversight that the USER must jump through hoops to locate their chat history and unpack it.  Um... did nobody think that was a good use case to include in the actual product?

**Reply by ontopoftheworldllc** [reply]
STILL dealing with this 11/8/2025.

**Comment by gregce** [comment]
Hi @TaihouKai and @Stabledog, to the extent its helpful... made a VSCode extension for GitHub Copilot that allows your to export chat history, share it and it automatically builds copilot instructions for you, check it out and let us know what you think: https://marketplace.visualstudio.com/items?itemName=SpecStory.specstory-vscode. 

Also, it will automatically autosave all of your chats to a `.specstory` directory in your project's root so you never have to "unpack" or manually save yourself.

**Reply by Wayne-Ca** [reply]
> Hi @TaihouKai and @Stabledog, to the extent its helpful... made a VSCode extension for GitHub Copilot that allows your to export chat history, share it and it automatically builds copilot instructions for you, check it out and let us know what you think: https://marketplace.visualstudio.com/items?itemName=SpecStory.specstory-vscode.
> 
> Also, it will automatically autosave all of your chats to a `.specstory` directory in your project's root so you never have to "unpack" or manually save yourself.

It would be helpful if Visual Studio Community 2022 could use this feature.

**Comment by Stabledog** [comment]
Thanks! I will have a look

**Comment by git45016683** [comment]
the extension is work to save chat history，but how can i import the history to the same project/code/floder in the other machine？？
